### DISCUSSION TITLE
Can't use Hubs between VLANs

### DISCUSSION THREAD

**Discussion by VCTGomes** [t0bst4r/home-assistant-matter-hub#272] [General]
I usually use IoT VLAN and my mainly VLAN, where HomeAssistant is installed. I have dozens of devices on IoT VLAN communicating with HomePods on mainly LAN, thankful to mDNS replicate and firewall.

However I can't replicate same with Alexa, that can't connect with Matter hub if Echos are on IoT LAN and Matter Hub on Mainly LAN.

Anyone knows what's wrong? Why can't Alexa connect through VLANs?

**Comment by t0bst4r** [comment]
Unfortunately I cannot help, because i don't have a VLAN setup and never needed to configure/maintain one.

If you find a way to get it working, please feel free to adjust the readme and add a FAQ.

**Reply by VCTGomes** [reply]
Ok, thanks!

Now I could confirm that. I migrated my HomeAssistant to IoT VLAN and it immediately stop working with my SmartThings and HomeKit, which can handle with Matter devices on IoT network.

So, indeed, for some reason Matter Hub Home Assistant can't work between VLANs.

**Reply by Apollon77** [reply]
Matter devices over HomeKit and SmartThings can work over VLANs? I doubt so tbh

**Reply by VCTGomes** [reply]
> Matter devices over HomeKit and SmartThings can work over VLANs? I doubt so tbh

Yeah. It works well, since you enabled mDNS reply

**Reply by Apollon77** [reply]
But you still state that Matter Hub is not working between VLANs? And yes MDNS is just one thing needed. The whole Matter communication is UDP which then also needs to be routed!
For "pure HomeKit" (not Matter!) just the discovery is done via MDNS but communication is TCP! But Matter relies on UDP

**Reply by VCTGomes** [reply]
> But you still state that Matter Hub is not working between VLANs? And yes MDNS is just one thing needed. The whole Matter communication is UDP which then also needs to be routed! For "pure HomeKit" (not Matter!) just the discovery is done via MDNS but communication is TCP! But Matter relies on UDP

Hm, ok. It doesn’t really matters, since firewall is correctly setup. In my case, all Matter devices are connected into IoT VLAN and HomePods and SmartThings are on LAN 1 and, even that, I have no problem… all devices works properly (except Matter hub)

**Reply by Apollon77** [reply]
Hm ... then it would interesting to find out what's the difference because there should be none. Matter Hub announces itself via MDNS and received UDP ... the same as other Matter devices .. if you find out whats different that would help

**Reply by tescophil** [reply]
Sounds like you need a mDNS reflector between your VLANS

**Reply by Apollon77** [reply]
@tescophil As said: MDNS is not enough ... needs to be UDP in general when

**Reply by tescophil** [reply]
No, it uses both.., from the Matter standard...

"Transport and IP Framing
This layer manages the transmission of the payload through the IP network to the peer device. It uses either the Transmission Control Protocol (TCP) or the combination of User Datagram Protocol (UDP) with Matter’s Message Reliability Protocol (MRP). The MRP implements retransmissions, provides confirmation of delivery, and ensures that duplicated messages are rejected."

**Reply by Apollon77** [reply]
hehe ... not really (currently) ... Spec and reality is a bit of a difference!
TCP was mainly added (aka "is partially no longerprovisional") in Matter 1.4 and will be used mainly for "large message" usecases like OTA updates and coming camera support or such.

The reality is that TCP is 100% optional for devices and (near to) no device on the market uses TCP at all. This means that Matter in reality as of now uses 100% UDP _only_.

**Comment by JTP335d** [comment]
I have a very similar setup with the same results. As a workaround I have home assistant on both networks. I run home assistant as a vm in proxmox and expose both vlans to it which shows up as two nics in home assistant. Home Assistant doesn't really support this though, and it is finicky because home assistant will pick a default and sometimes a reboot will change the default to the IoT and my Apple integration will fail to load. I don't want my Echos on the same vlan as my home assistant. A better option might be to run Matter Hub separate from Home assistant but I haven't explored that.
